1. Examining childcare data
I start by importin a new module called geopandas to my notebook.


In [2]:
import geopandas as gpd

Next, I need to import data. Here, I have downloaded data on childcare and familychild care from Department of Social Services.
https://www.ccld.dss.ca.gov/carefacilitysearch/DownloadData. The data is in csv format. 
But, I first want to know where my file is located. 


In [3]:
pwd

'/home/jovyan/up206a/Weeks/Week02/Childcare data'

Now I use relative path to point to where data for childcare is located using :/

In [4]:
childcare = gpd.read_file('../Childcare data/childcarecenters.csv')

Next, I want to know data type for childcare data. 

In [5]:
type(childcare)

geopandas.geodataframe.GeoDataFrame

But, what does my data actually even look like?

In [8]:
childcare.head()

Facility Type Facility Number  \
0  DAY CARE CENTER       197416900   
1  DAY CARE CENTER       191607790   
2  DAY CARE CENTER       197416698   
3  DAY CARE CENTER       197493820   
4  DAY CARE CENTER       384001195   

                                     Facility Name  \
0  107TH. STREET ELEMENTARY SCHOOL CSPP/HEAD START   
1                            10TH STREET PRESCHOOL   
2              186TH STREET ELEMENTARY SCHOOL CSPP   
3                  1ST CLASS PREPARATORY PRESCHOOL   
4                                1ST PLACE 2 START   

                          Licensee Facility Administrator  \
0   LAUSD EARLY CHILDHOOD EDUCATON        RIOS, REUBEN M.   
1                WASSON, CINDY LEE          WASSON, CINDY   
2  LAUSD/EARLY CHILDHOOD EDUCATION        REED, MARCIA S.   
3            1ST CLASS PREPARATORY       MARRISCHIA DAVIS   
4                1ST PLACE 2 START           SANDRA DAVIS   

  Facility Telephone Number                Facility Address  Facility City  \
0            (323) 756-8137  146 EAST 107TH. STREET ROOM K4    LOS ANGELES   
1            (310) 458-4088                1444 10TH STREET   SANTA MONICA   
2            (310) 324-1153   1581 WEST 186TH STREET ROOM 9        GARDENA   
3            (310) 925-6394                   3459 MC MANUS    CULVER CITY   
4            (415) 333-2659              1252 SUNNYDALE AVE  SAN FRANCISCO   

  Facility State Facility Zip  ...               POC Dates  \
0             CA        90003  ...                           
1             CA        90401  ...              08/29/2019   
2             CA        90248  ...                           
3             CA        90232  ...                           
4             CA        94134  ...  08/17/2016, 08/17/2016   

                                  All Visit Dates  \
0              03/26/2019, 03/26/2019, 05/06/2016   
1  07/29/2019, 01/25/2018, 03/11/2016, 03/11/2016   
2                                      12/13/2016   
3                                      10/11/2018   
4  09/12/2019, 10/24/2017, 08/24/2016, 08/03/2016   

               Inspection Visit Dates Inspect TypeA Inspect TypeB  \
0              03/26/2019, 05/06/2016             0             0   
1              07/29/2019, 01/25/2018             0             1   
2                          12/13/2016             0             0   
3                                                 0             0   
4  09/12/2019, 10/24/2017, 08/03/2016             0             2   

  Other Visit Dates Other TypeA Other TypeB  \
0        03/26/2019           0           0   
1                             0           0   
2                             0           0   
3        10/11/2018           0           0   
4        08/24/2016           0           0   

  Complaint Info- Date, #Sub Aleg, # Inc Aleg, # Uns Aleg, # TypeA, # TypeB ...  \
0                                      No Complaints                              
1                                         03/14/2016                              
2                                      No Complaints                              
3                                      No Complaints                              
4                                      No Complaints                              

  geometry  
0     None  
1     None  
2     None  
3     None  
4     None  

[5 rows x 32 columns]

Let's see a sample of our dataset

In [9]:
childcare.sample()

Facility Type Facility Number      Facility Name  \
8464  DAY CARE CENTER       343622733  NICHOLS PRESCHOOL   

                          Licensee Facility Administrator  \
8464  LODI UNIFIED SCHOOL DISTRICT       LOPEZ, MARY LISA   

     Facility Telephone Number            Facility Address Facility City  \
8464            (209) 331-7378  1301 SOUTH CRESCENT AVENUE          LODI   

     Facility State Facility Zip  ... POC Dates         All Visit Dates  \
8464             CA        95240  ...            12/20/2019, 10/09/2018   

     Inspection Visit Dates Inspect TypeA Inspect TypeB Other Visit Dates  \
8464             12/20/2019             0             0        10/09/2018   

     Other TypeA Other TypeB  \
8464           0           0   

     Complaint Info- Date, #Sub Aleg, # Inc Aleg, # Uns Aleg, # TypeA, # TypeB ...  \
8464                                      No Complaints                              

     geometry  
8464     None  

[1 rows x 32 columns]

Now that I know what variables are in this dataset, let's see what data types I have.

In [10]:
childcare.dtypes

Facility Type                                                                      object
Facility Number                                                                    object
Facility Name                                                                      object
Licensee                                                                           object
Facility Administrator                                                             object
Facility Telephone Number                                                          object
Facility Address                                                                   object
Facility City                                                                      object
Facility State                                                                     object
Facility Zip                                                                       object
County Name                                                                        object
Regional O

There def was an even better command to get this info. It is the info command

In [11]:
childcare.info()

<class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 19005 entries, 0 to 19004
Data columns (total 32 columns):
 #   Column                                                                         Non-Null Count  Dtype   
---  ------                                                                         --------------  -----   
 0   Facility Type                                                                  19005 non-null  object  
 1   Facility Number                                                                19005 non-null  object  
 2   Facility Name                                                                  19005 non-null  object  
 3   Licensee                                                                       19005 non-null  object  
 4   Facility Administrator                                                         19005 non-null  object  
 5   Facility Telephone Number                                                      19005 non-null  object  
 6   Facili

So, most of our data type is object, aka string. That's a pretty big dataset. But, how big. How many rows and columns are there?

In [12]:
childcare.shape

(19005, 32)

32 columns! 19,005 rows. But, that's too many columns. I definitely do not need information on inspection visits, complaint visits, and all of that. Let's trim our data. 

In [15]:
childcare[['Facility Type', 'Facility Number', 'Facility Name', 'Licensee', 'Facility Administrator','Facility Telephone Number', 'Facility Address','Facility City ','Facility State', 'Facility Zip', 'County Name','Facility Capacity'   ]]

KeyError: "['Facility City '] not in index"